tree/treeplayer/inc/ROOT/TVec.hxx

```diff
@@ -0,0 +1,574 @@
+   }
+
+   template <typename V>
+   TVec<T> &operator-=(const V &c)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, [&c](const T &t) { return t - c; });
+   }
+
+   template <typename V>
+   TVec<T> &operator*=(const V &c)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, [&c](const T &t) { return t * c; });
+   }
+
+   template <typename V>
+   TVec<T> &operator/=(const V &c)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, [&c](const T &t) { return t / c; });
+   }
+
+   template <typename V>
+   TVec<T> &operator%=(const V &c)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, [&c](const T &t) { return t % c; });
+   }
+
+   template <typename V>
+   TVec<T> &operator+=(const TVec<V> &v0)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, v0, "+", [](const T &t, const V &v) { return t + v; });
+   }
+
+   template <typename V>
+   TVec<T> &operator-=(const TVec<V> &v0)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, v0, "-", [](const T &t, const V &v) { return t - v; });
+   }
+
+   template <typename V>
+   TVec<T> &operator*=(const TVec<V> &v0)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, v0, "*", [](const T &t, const V &v) { return t * v; });
+   }
+
+   template <typename V>
+   TVec<T> &operator/=(const TVec<V> &v0)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, v0, "/", [](const T &t, const V &v) { return t / v; });
+   }
+
+   template <typename V>
+   TVec<T> &operator%=(const TVec<V> &v0)
+   {
+      return ROOT::Internal::VecOps::OperateInPlace(*this, v0, "%", [](const T &t, const V &v) { return t % v; });
+   }
+};
+
+/** @name Math Operators with scalars
+ *  Math operators involving TVec
+*/
+///@{
+template <typename T, typename V>
+auto operator+(const TVec<T> &v, const V &c) -> TVec<decltype(v[0] + c)>
+{
+   return ROOT::Internal::VecOps::Operate(v, [&c](const T &t) { return t + c; });
+}
+
+template <typename T, typename V>
+auto operator-(const TVec<T> &v, const V &c) -> TVec<decltype(v[0] - c)>
+{
+   return ROOT::Internal::VecOps::Operate(v, [&c](const T &t) { return t - c; });
+}
+
+template <typename T, typename V>
+auto operator*(const TVec<T> &v, const V &c) -> TVec<decltype(v[0] * c)>
+{
+   return ROOT::Internal::VecOps::Operate(v, [&c](const T &t) { return t * c; });
+}
+
+template <typename T, typename V>
+auto operator/(const TVec<T> &v, const V &c) -> TVec<decltype(v[0] / c)>
+{
+   return ROOT::Internal::VecOps::Operate(v, [&c](const T &t) { return t / c; });
+}
+
+template <typename T, typename V>
+auto operator%(const TVec<T> &v, const V &c) -> TVec<decltype(v[0] % c)>
+{
+   return ROOT::Internal::VecOps::Operate(v, [&c](const T &t) { return t % c; });
+}
+
+using Boolean_t = int;
+
+template <typename T, typename V>
+auto operator>(const TVec<T> &v, const V &c) -> decltype(v[0] > c, TVec<Boolean_t>())
```

Comment:
What about `op>(V, TVec<T>)`? (Possibly also for all the others.)

Reply:
Good point. It's odd to implement non-commutative arithmetic ;-)